src/loader.c

```diff
@@ -584,7 +580,7 @@ static void unquote_add_authfile_user(const char *username, const char *password
 	copy_quoted(real_user, username, sizeof(real_user));
 	copy_quoted(real_passwd, password, sizeof(real_passwd));
 
-	user = add_global_user(real_user, real_passwd);
+	user = add_new_global_user(real_user, real_passwd);
```

Comment:
this one i'm not completely sure what the desired behaviour is. the old `add_global_user` is essentially a find + add. but most other places in the code use `find_global_user` + `add_global_user`. so the intention is not too clear to me.

is it better to use this function here or use the new `find_or_add_new_global_user`?